2_Single channel display/Python_GUI_day2_simple.py

```diff
@@ -1,30 +1,33 @@
 import dearpygui.dearpygui as dpg
-# math is used to generate data for the plot widget
-from math import cos
+# numpy is used to generate data for the plot widget
+import numpy as np
 
 global SERIAL_TAG
 
 # The following function is used to generate and update the plot widget
 def update_series(j):
-    cosdatax = []
-    cosdatay = []
-    # when you display data dynamically, you need to specify the length of the data
-    # in this case, the length of the data is 500, that is, there are 500 points in the plot widget
-    for i in range(0, 500):
-        cosdatax.append(i / 1000)
-        cosdatay.append(0.5 + 0.5 * cos(50 * (i+j) / 1000))
+    """update data for plotting
+    This function is used to update demo data for plotting.
+    The data is generated using the following equation:
+    y = 0.5 + 0.5 * sin(50 * x)
+    when you display data dynamically, you need to specify the length of the data
+    in this case, the length of the data is 500, that is, there are 500 points in the plot widget
+    """
+    cosdatax = np.arange(500)/1000
+    cosdatay = 0.5 + 0.5 * np.cos(50 * (cosdatax+j/ 1000))
     """
     Pay attention
-    When you plot data without GUI, you can use plt.plot(data)
     When you plot data with GUI, you need to use dpg.set_value() to update the data, the data will be shown
-    in the plot widget with the tag "series_tag", this tag is assigned to the plot widget at line 41.
+    in the plot widget with the tag "series_tag", this tag is assigned to the plot widget at line 46.
     """
-    dpg.set_value(SERIAL_TAG, [cosdatax, cosdatay])
+    dpg.set_value(SERIAL_TAG, [list(cosdatax), list(cosdatay)])
```

Comment:
Why is it necessary to convert the arrays to lists here, but `dpg.add_line_series` below works well directly with arrays? Is this conversion really necessary?